### DISCUSSION TITLE
Displaying error messages

### DISCUSSION THREAD

**Discussion by josephcrawfordSRH** [johnlindquist/kit#1379] [Q&A] [answered]
Is there a way to display error messages for instance when someone tries to use my script without having Git installed I would want to stop the process and display an error that git is required.  I know in this case I could do a number of things such as check for homebrew or some other installer and ask to install but then there is a lot of logic branching that I would want to avoid.

Currently I just want to display an error message in Script Kit or something

**Comment by johnlindquist** [comment, marked as answer]
@josephcrawfordSRH 

Error messages should be improved in the 2.0 release candidate:

https://github.com/johnlindquist/kitapp/releases/tag/v1.99.79

(Using "grit" instead of "git" for testing purposes)

```js
let hasGrit = await isBin("grit")
if (hasGrit) {
  // Do grit stuff
} else {
  throw Error("You need to install grit")
}
```

Or if you want to provide much more info:

```js

let hasGrit = await isBin("grit")
if (hasGrit) {
  // Do grit stuff
} else {
  let message = md(`# Grit Not Found
    
The \`grit\` command was not found on your PATH accessible in this script.

Please check \`grit\` is installed.

Options:
- Add PATH variable in ~/.kenv/.env
- Modify this script to use the full path to \`grit\`

Click [here](https://google.com) to learn more about grit.
    `)
  await div(message)
}
```

![CleanShot 2023-11-11 at 07 46 32](https://github.com/johnlindquist/kit/assets/36073/fce998ba-6cbf-4ad9-8dfa-6e67fc7cecc1)

**Reply by josephcrawfordSRH** [reply]
oh very nice, thank you!
